app/database/group_group_store_ancestors_integration_test.go

```diff
@@ -145,10 +158,103 @@ func TestGroupGroupStore_CreateNewAncestors_ProcessesOnlyDirectRelationsOrAccept
 	assert.NoError(t, groupGroupStore.GroupAncestors().Order("child_group_id, ancestor_group_id").Scan(&result).Error())
 
 	assert.Equal(t, []groupAncestorsResultRow{
-		{ChildGroupID: 1, AncestorGroupID: 1, IsSelf: true},
-		{ChildGroupID: 2, AncestorGroupID: 2, IsSelf: true},
-		{ChildGroupID: 3, AncestorGroupID: 3, IsSelf: true},
-		{ChildGroupID: 4, AncestorGroupID: 4, IsSelf: true},
+		{ChildGroupID: 1, AncestorGroupID: 1, IsSelf: true, ExpiresAt: maxExpiresAt},
+		{ChildGroupID: 2, AncestorGroupID: 2, IsSelf: true, ExpiresAt: maxExpiresAt},
+		{ChildGroupID: 3, AncestorGroupID: 3, IsSelf: true, ExpiresAt: maxExpiresAt},
+		{ChildGroupID: 4, AncestorGroupID: 4, IsSelf: true, ExpiresAt: maxExpiresAt},
+	}, result)
+
+	var propagateResult []groupPropagateResultRow
+	assert.NoError(t, groupGroupStore.Table("groups_propagate").Order("id").Scan(&propagateResult).Error())
+	assert.Equal(t, []groupPropagateResultRow{
+		{ID: 1, AncestorsComputationState: "done"},
+		{ID: 2, AncestorsComputationState: "done"},
+		{ID: 3, AncestorsComputationState: "done"},
+		{ID: 4, AncestorsComputationState: "done"},
+	}, propagateResult)
+}
+
+func TestGroupGroupStore_CreateNewAncestors_PropagatesExpiresAt(t *testing.T) {
+	db := testhelpers.SetupDBWithFixture("group_group_store/ancestors/_common")
+	defer func() { _ = db.Close() }()
+
+	groupGroupStore := database.NewDataStore(db).GroupGroups()
+	assert.NoError(t, groupGroupStore.Where("parent_group_id=1 AND child_group_id=2").
+		UpdateColumn("expires_at", "3019-12-31 20:10:30").Error())
+	assert.NoError(t, groupGroupStore.Where("parent_group_id=1 AND child_group_id=3").
+		UpdateColumn("expires_at", "3019-10-20 14:13:55").Error())
+	assert.NoError(t, groupGroupStore.Where("parent_group_id=1 AND child_group_id=4").
+		UpdateColumn("expires_at", "3020-09-15 21:13:59").Error())
+	assert.NoError(t, groupGroupStore.Where("parent_group_id=2 AND child_group_id=3").
+		UpdateColumn("expires_at", "3020-10-20 10:23:40").Error())
+	assert.NoError(t, groupGroupStore.Where("parent_group_id=2 AND child_group_id=4").
+		UpdateColumn("expires_at", "3021-08-10 12:11:45").Error())
+	assert.NoError(t, groupGroupStore.Where("parent_group_id=3 AND child_group_id=4").
+		UpdateColumn("expires_at", "3022-05-11 17:43:24").Error())
```

Comment:
I don't get why you make vary the year, month, day, hour, min, sec. (anything to be validated really?). I would expect only one varying (for instance all different years and that's all) to be able to save 80% of time in trying to verify something here. 🙂
Or maybe I missed some interesting hidden logic 🤔 ... actually I don't care the day,  etc changes, but at least they should all be on different years so that we do not have to check year and month together in the mapping. 

But that looks correct at the end, I just have a headache now 😛

Reply:
fixed

Reply:
much better :-)